### ISSUE TITLE
BOLD: transformation lambda function

### ISSUE THREAD

**Issue by caglorithm** [neurolib-dev/neurolib#57]
The BOLD model gets input from various different sources. For now, models which don't ouput a firing rate in `Hz` can specify `normalize_bold_input = True` to normalize their output (via `normalize_bold_input_max = 50`) with a very simple predefined transformation that is baked into the BOLD model. This leads to some errors, specifically for models with negative outputs and non-oscillating states where there is no real amplitude to scale. 

Therefore, it might be a better option to define a model-specific transformation function of the outputs such that the BOLD model can work with them. I am thinking something along the lines of 
```python 
model.boldTransorm = (lambda x: (x + 50)*2.5)
```

and initialize the BOLD model in `neurolib/models/model.py` with
```python
bold.BOLDModel(self.params["N"], self.params["dt"], self.boldTransorm)
```

In `run()` in `model/bold/model.py`, the input would then be passed through this transformation function before passed to the bold timeintegration.

This way, every model would have the freedom of transforming their output to whatever they want the BOLD model to get as an input.

Thoughts?

**Comment by ChristophMetzner** [comment]
That sounds like a good idea. I guess, for many models that have been around for a while, people have already used transfer functions to produce firing rate output, which could then simply be used. This would also match the behaviour of the model in neurolib with the behaviour of the same model outside of neurolib. For example, someone reads papers about model A (for which transfer function f_A is traditionally being used) and wants to use it in neurolib. However, they are not aware of the different output scaling in neurolib right now. This could cause confusion and even wrong conclusions. The proposed change would prevent that.

**Comment by jajcayn** [comment]
+1 for the user-defined `lambda` functions. for identity (i.e. no transform), it can be just `lambda x: x`

**Comment by caglorithm** [comment]
> People have already used transfer functions to produce firing rate output ... However, they are not aware of the different output scaling in neurolib right now

Right now, the output itself is not scaled before being "recorded" to the `model.output` object. This could be done in a very similar way but also done by the model integration itself. `aln` for example is originally implemented to output `kHz` but I convert it to `Hz` in a very hackish way (see Issue #53 ...). I don't quite like it I must admit because the `aln` model also expects `kHz` *input*. So once you start transforming outputs, I think you might also have to start transforming inputs...

**Comment by caglorithm** [comment]
Thanks for the input to both of you. I will make a PR on this, stay tuned!

**Comment by caglorithm** [comment]
Closed by #59
